src/components/app.tsx

```diff
@@ -6,51 +6,70 @@ import { BaseComponent, IBaseProps } from "./base";
 import { urlParams } from "../utilities/url-params";
 
 import "./app.sass";
+import { GroupChooserComponent } from "./group-chooser";
+import { observable } from "mobx";
+import { IStores } from "../models/stores";
 
 interface IProps extends IBaseProps {}
 
+export const authAndConnect = (stores: IStores) => {
+  const {appMode, user, db, ui} = stores;
+
+  authenticate(appMode, urlParams.token, urlParams.domain)
```

Comment:
I made this comment before, but it seems like this could/should move into an authentication-specific file.